File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/cirptlab/i/ICirptExtService.java
package iih.ci.ord.cirptlab.i;

import xap.mw.core.data.BizException;
import iih.ci.ord.cirptlab.d.CirptlabAggDO;
import iih.ci.ord.cirptobs.d.CiRptObsDO;
import iih.ci.ord.cirptpathgy.d.CiRptPathgyDO;

/**
 * @author 作者 :huang_junhao
 * @version 创建时间：2016年11月18日 下午7:54:57
 * 类说明：检查、检验结果录入
 */
public interface ICirptExtService {

	/**
	 * 检查报告结果录入
	 * @param ciRptObsDOs
	 * @return
	 * @throws BizException
	 */
	public abstract CiRptObsDO[] saveCiRptObsDO(CiRptObsDO[] ciRptObsDOs) throws BizException;
	
	/**
	 * 检验报告结果录入
	 * @param cirptlabAggDOs
	 * @return
	 * @throws BizException
	 */
	public abstract CirptlabAggDO saveCirptlabAggDO(CirptlabAggDO cirptlabAggDOs) throws BizException;
	
	/**
	 * 病理报告结果录入
	 * @param pathgyDOs
	 * @return
	 * @throws BizException
	 */
	public abstract CiRptPathgyDO[] saveCiRptPathgyDO(CiRptPathgyDO[] pathgyDOs) throws BizException;
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/cirptlab/i/ICirptExtService.cs
/* 文件头注释
======================================================================
* Filename: PaintToolKit
* Date: 2016-11-18 20:22:34
* Compiler: Visual Studio 2010
* Author: huang_junhao
* Company: Copyright 2016 by PKU healthcare IT Co.,Ltd.
* Description: 检查、检验报告结果录入保存
======================================================================
*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.ci.ord.cirptlab.d;
using iih.ci.ord.cirptobs.d;
using iih.ci.ord.cirptpathgy.d;

namespace iih.ci.iih.ci.ord.cirptlab.i
{
    public interface ICirptExtService
    {
    
        CiRptObsDO[] saveCiRptObsDO(CiRptObsDO[] ciRptObsDOs);

        CirptlabAggDO saveCirptlabAggDO(CirptlabAggDO cirptlabAggDOs);

        CiRptPathgyDO[] saveCiRptPathgyDO(CiRptPathgyDO[] pathgyDOs);
    }
}
